File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        if(nums.size()==1 or nums.back()>nums[0]) return nums[0];
        int low = 0, high = nums.size()-1;
        while(low<high){
            int mid = low +(high-low)/2;
            if(nums[mid]>nums.back())low = mid+1;
            else high = mid;
        }
        return nums[low];
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-4.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        def go(mid):
            needed_time = 0
            for x in piles:
                needed_time += (x//mid)
                if x%mid:
                    needed_time +=1
            return needed_time <=h
        low, high = 1, max(piles)
        while low < high:
            mid = low + (high-low)//2
            if (go(mid)):
                high = mid
            else :
                low = mid+1
        return low